conans/client/installer.py

```diff
@@ -454,3 +456,15 @@ def _call_package_info(self, conanfile, package_folder, is_editable):
                 self._hook_manager.execute("post_package_info", conanfile=conanfile)
 
         conanfile.cpp_info.check_component_requires(conanfile)
+
+    def _call_install_method(self, conanfile, install_folder):
+        if hasattr(conanfile, "install"):
+            conanfile.folders.set_install_folder(install_folder)
+            if not os.path.exists(install_folder):
+                mkdir(install_folder)
+                conanfile.output.highlight("Calling install()")
+                with conanfile_exception_formatter(conanfile, "install"):
+                    with conanfile_remove_attr(conanfile, ['cpp_info', 'settings', 'options'], 'install'):
+                        conanfile.install()
+            conanfile.folders.set_base_package(install_folder)
```

Comment:
The feature works by replacing the package folder so that downstream users transparently use the installed version instead of the immutable one